### PR TITLE
Implement #104: add getObjectPath() for classes

### PR DESCRIPTION
**Pull request by ChristianS99**
is @brief enough or should something more added?

### CODE REVIEW

File: include/sdbus-c++/AdaptorInterfaces.h [hunk 1]
```diff
@@ -107,6 +107,14 @@ namespace sdbus {
         {
         }
 
+        /*!
+         * @brief return object path of the underlying DBus object
+         */
+        const std::string& getObjectPath() const
```
Comment (sangelovic) [inline include/sdbus-c++/AdaptorInterfaces.h:113]: I'd prefer that we define this function at the end of the class, after `unregisterAdaptor`. This method is not so essential as the other ones for class operation, and these ones I would like to keep up in the class to be visible immediately. Would you please move it there?

File: include/sdbus-c++/AdaptorInterfaces.h [hunk 2]
```diff
@@ -107,6 +107,14 @@ namespace sdbus {
         {
         }
 
+        /*!
+         * @brief return object path of the underlying DBus object
```
[the remark this answers is not in the document]
Comment (sangelovic) [reply include/sdbus-c++/AdaptorInterfaces.h:111]: This applies also to all other comments in this pull request...

File: include/sdbus-c++/IObject.h [hunk 3]
```diff
@@ -61,6 +61,11 @@ namespace sdbus {
     public:
         virtual ~IObject() = default;
 
+        /*!
+         * @brief return object path of the underlying DBus object
+         */
+        virtual const std::string& getObjectPath() const = 0;
```
Comment (sangelovic) [inline include/sdbus-c++/IObject.h:67]: Would you please move the declaration of the function to the bottom of the class? This also has the advantage that the binary interface will remain backwards compatible with this change.

File: include/sdbus-c++/IProxy.h [hunk 4]
```diff
@@ -65,6 +65,11 @@ namespace sdbus {
     public:
         virtual ~IProxy() = default;
 
+        /*!
+         * @brief return object path of the underlying DBus object
+         */
+        virtual const std::string& getObjectPath() const = 0;
```
Comment (sangelovic) [inline include/sdbus-c++/IProxy.h:71]: Ditto, moving to the bottom of the class...

File: include/sdbus-c++/ProxyInterfaces.h [hunk 5]
```diff
@@ -141,6 +141,14 @@ namespace sdbus {
         {
         }
 
+        /*!
+         * @brief return object path of the underlying DBus object
+         */
+        const std::string& getObjectPath() const
+        {
+            return getProxy().getObjectPath();
+        }
```
Comment (sangelovic) [inline include/sdbus-c++/ProxyInterfaces.h:150]: Ditto, moving to the bottom of the class...

File: src/Object.h [hunk 6]
```diff
@@ -45,6 +45,11 @@ namespace sdbus::internal {
     public:
         Object(sdbus::internal::IConnection& connection, std::string objectPath);
 
+        const std::string& getObjectPath() const override
+        {
+            return objectPath_;
+        }
```
Comment (sangelovic) [inline src/Object.h:51]: The order of methods here doesn't matter much already, but I would also move it after all other virtual methods to keep ordering consistency.

Comment (sangelovic) [inline src/Object.h:51]: Additionally, would you please declare the method here and implement it in the `.cpp` file? The method is virtual, so inlining it doesn't truly make sense, the method will be instantiated in several object files and then the linker will anyway have to pick one single place. And anyway, defining it in the `.cpp` file will be consistent with other interface, virtual methods.

File: src/Proxy.h [hunk 7]
```diff
@@ -50,6 +50,11 @@ namespace sdbus::internal {
              , std::string destination
              , std::string objectPath );
 
+        const std::string& getObjectPath() const override
+        {
+            return objectPath_;
+        }
```
Comment (sangelovic) [inline src/Proxy.h:56]: Ditto, defining the method in the `.cpp` file, see comment above...

File: include/sdbus-c++/AdaptorInterfaces.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChristianS99) [reply include/sdbus-c++/AdaptorInterfaces.h]: Done

File: include/sdbus-c++/AdaptorInterfaces.h (hunk 1 above)
Comment (ChristianS99) [reply include/sdbus-c++/AdaptorInterfaces.h]: Done

File: include/sdbus-c++/IObject.h (hunk 3 above)
Comment (ChristianS99) [reply include/sdbus-c++/IObject.h]: Done

File: include/sdbus-c++/IProxy.h (hunk 4 above)
Comment (ChristianS99) [reply include/sdbus-c++/IProxy.h]: Done

File: src/Object.h (hunk 6 above)
Comment (ChristianS99) [reply src/Object.h]: Done

File: include/sdbus-c++/ProxyInterfaces.h (hunk 5 above)
Comment (ChristianS99) [reply include/sdbus-c++/ProxyInterfaces.h]: Done

File: src/Object.h (hunk 6 above)
Comment (ChristianS99) [reply src/Object.h]: Done

File: src/Proxy.h (hunk 7 above)
Comment (ChristianS99) [reply src/Proxy.h]: Done

**Pull request merged**